scripts/build/termux_download.sh

```diff
@@ -1,53 +1,68 @@
 #!/usr/bin/bash
 
 termux_download() {
-	if [ $# != 3 ]; then
+	if [[ $# != 3 ]]; then
 		termux_error_exit "termux_download(): Invalid arguments - expected \$URL \$DESTINATION \$CHECKSUM"
 	fi
 	local URL="$1"
 	local DESTINATION="$2"
 	local CHECKSUM="$3"
 
-	if [ -f "$DESTINATION" ] && [ "$CHECKSUM" != "SKIP_CHECKSUM" ]; then
+	if [[ -f "$DESTINATION" ]] && [[ "$CHECKSUM" != "SKIP_CHECKSUM" ]]; then
 		# Keep existing file if checksum matches.
 		local EXISTING_CHECKSUM
-		EXISTING_CHECKSUM=$(sha256sum "$DESTINATION" | cut -f 1 -d ' ')
-		if [ "$EXISTING_CHECKSUM" = "$CHECKSUM" ]; then return; fi
+		EXISTING_CHECKSUM=$(sha256sum "$DESTINATION" | cut -d' ' -f1)
+		[[ "$EXISTING_CHECKSUM" == "$CHECKSUM" ]] && return
 	fi
 
 	local TMPFILE
 	local -a CURL_OPTIONS=(
 		--fail
-		--retry 20
+		--retry 10
 		--retry-connrefused
-		--retry-delay 30
+		--retry-delay 5
+		--speed-limit 10000
+		--speed-time 30
 		--location
 	)
 	TMPFILE=$(mktemp "$TERMUX_PKG_TMPDIR/download.${TERMUX_PKG_NAME-unnamed}.XXXXXXXXX")
-	if [ "${TERMUX_QUIET_BUILD-}" = true ]; then
+	if [[ "${TERMUX_QUIET_BUILD-}" == "true" ]]; then
 		CURL_OPTIONS+=( --no-progress-meter)
 	fi
 
 	echo "Downloading ${URL}"
-	if curl "${CURL_OPTIONS[@]}" --output "$TMPFILE" "$URL"; then
-		local ACTUAL_CHECKSUM
-		ACTUAL_CHECKSUM=$(sha256sum "$TMPFILE" | cut -f 1 -d ' ')
-		if [ "$CHECKSUM" != "SKIP_CHECKSUM" ]; then
-			if [ "$CHECKSUM" != "$ACTUAL_CHECKSUM" ]; then
-				>&2 printf "Wrong checksum for %s\nExpected: %s\nActual:   %s\n" \
-					   "$URL" "$CHECKSUM" "$ACTUAL_CHECKSUM"
-				return 1
+	if ! curl "${CURL_OPTIONS[@]}" --output "$TMPFILE" "$URL"; then
+		local e=1
```

Comment:
"Well-chosen identifiers make it significantly easier for developers" - https://en.wikipedia.org/wiki/Naming_convention_(programming)

Reply:
Yep, `err` or `error` would probably be preferable.